.github/workflows/playwright.yml

```diff
@@ -0,0 +1,84 @@
+name: Playwright Tests
+on:
+  push:
+    branches-ignore: ['main']
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - name: Install Packages
+        run: npm ci
+      - name: Build tsoa
+        run: npm run tsoa:build
+      - name: Build
+        run: npm run build
+      - name: touch env
+        run: touch .env
+      - name: Setup dependencies
+        run: docker compose up -d
+      - name: Sleep
+        uses: kibertoad/wait-action@1.0.1
+        with:
+          time: '30s'
+      - name: Run database migrations
+        run: npm run db:migrate
+        env:
+          NODE_ENV: test
+      - name: Start server in detached mode
+        run: nohup npm run start & # Starts the server in detached mode
+        env:
+          PORT: 3000
+          LOG_LEVEL: debug
+          DB_HOST: localhost
+          DB_NAME: veritable-ui
+          DB_USERNAME: postgres
+          DB_PASSWORD: postgres
+          DB_PORT: 5432
+          COOKIE_SESSION_KEYS: '["secret"]'
+          PUBLIC_URL: http://localhost:3000
+          IDP_CLIENT_ID: veritable-ui
+          IDP_PUBLIC_URL_PREFIX: http://localhost:3080/realms/veritable/protocol/openid-connect
+          IDP_INTERNAL_URL_PREFIX: http://localhost:3080/realms/veritable/protocol/openid-connect
+          CLOUDAGENT_ADMIN_ORIGIN: http://localhost:3100
+          CLOUDAGENT_ADMIN_WS_ORIGIN: ws://localhost:3100
+          INVITATION_PIN_SECRET: secret
+          INVITATION_FROM_COMPANY_NUMBER: 07964699
+          ISSUANCE_DID_POLICY: EXISTING_OR_NEW
+          ISSUANCE_SCHEMA_POLICY: EXISTING_OR_NEW
+          ISSUANCE_CRED_DEF_POLICY: EXISTING_OR_NEW
+          COMPANY_PROFILE_API_KEY: apikey
+      - name: Install Playwright Browsers
+        run: npm run playwright:install
+      - name: Run Playwright tests
+        run: npm run playwright:test
+      - uses: actions/upload-artifact@v4
+        if: always()
```

Comment:
not sure if I understand the usage of `if: always()`

Reply:
it ensures the step runs regardless of whether previous steps have succeeded or failed. Not necessary, but we want to always upload the report, even if the tests fail.... I think

Reply:
It does, otherwise normally a workflow will end after a failed step.